s1tbx-op-insar/src/main/java/org/esa/s1tbx/insar/gpf/support/CRSGeoCodingHandler.java

```diff
@@ -63,13 +76,26 @@ public CRSGeoCodingHandler(final Product sourceProduct, final String mapProjecti
         bounds.setFrameFromDiagonal(lonMin, srcImageBoundary.latMin, lonMax, srcImageBoundary.latMax);
         final ReferencedEnvelope boundsEnvelope = new ReferencedEnvelope(bounds, DefaultGeographicCRS.WGS84);
         final ReferencedEnvelope targetEnvelope = boundsEnvelope.transform(targetCRS, true, 200);
-        targetWidth = (int) Math.floor(targetEnvelope.getSpan(0) / pixelSizeX);
-        targetHeight = (int) Math.floor(targetEnvelope.getSpan(1) / pixelSizeY);
+
+        double easting = targetEnvelope.getMinimum(0);
+        double northing = targetEnvelope.getMaximum(1);
+        if (alignToStandardGrid) {
+            // Force pixels to be aligned with a specified origin point (e.g. 0,0) in the output CRS.
+            // This guarantees that the image grids are always aligned when reprojecting or resampling images.
+            easting = Math.floor((easting - standardGridOriginX) / pixelSizeX) * pixelSizeX + standardGridOriginX;
+            northing = Math.ceil((northing - standardGridOriginY) / pixelSizeY) * pixelSizeY + standardGridOriginY;
+            targetWidth = (int) Math.ceil((targetEnvelope.getMaximum(0) - easting) / pixelSizeX);
+            targetHeight = (int) Math.ceil((northing - targetEnvelope.getMinimum(1)) / pixelSizeY);
```

Comment:
The existing code uses `floor()` to calculate the image dimensions, but `ceil()` seemed a bit more conservative and appropriate to me since it creates one more output sample right after the last input sample.